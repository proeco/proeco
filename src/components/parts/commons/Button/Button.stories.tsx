import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Button } from './Button';

export default {
  title: 'parts/commons/Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => (
  <div className="d-flex gap-3">
    <div>
      <Button size="sm" {...args}>
        Button!
      </Button>
    </div>
    <div>
      <Button {...args}>Button!</Button>
    </div>
    <div>
      <Button size="lg" {...args}>
        Button!
      </Button>
    </div>
  </div>
);

export const Contained = Template.bind({});
Contained.args = {
  color: 'primary',
  disabled: false,
  outlined: true,
};

export const Outlined = Template.bind({});
Outlined.args = {
  color: 'primary',
  outlined: true,
  disabled: false,
};
