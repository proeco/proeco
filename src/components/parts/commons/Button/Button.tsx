import React, { FC } from 'react';

type Props = {
  disabled?: boolean;
  size?: 'sm' | 'lg';
  color: 'primary' | 'info';
  outlined?: boolean;
};

export const Button: FC<Props> = ({ children, disabled, size, color, outlined }) => {
  const classNames = ['btn'];
  if (disabled) classNames.push('disabled');
  if (size) classNames.push(`btn-${size}`);
  if (outlined) {
    classNames.push(`btn-${color}`);
  } else {
    classNames.push(`btn-outline-${color}`);
  }

  return (
    <button type="button" className={classNames.join(' ')}>
      {children}
    </button>
  );
};
