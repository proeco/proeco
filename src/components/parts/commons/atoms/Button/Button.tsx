import React, { VFC, ComponentProps } from 'react';
import { styled } from '@mui/material/styles';
import { Button as MuiButton } from '@mui/material';

type Bold = {
  bold?: boolean;
};

type Props = ComponentProps<typeof MuiButton> & Bold;

export const Button: VFC<Props> = ({ bold, ...rest }) => {
  return <StyledButton {...rest} bold={bold ? 1 : 0} />;
};

const StyledButton = styled(MuiButton)<{ bold: number }>`
  &.MuiButton-containedSecondary,
  &.MuiButton-containedPrimary,
  &.MuiButton-containedError,
  &.MuiButton-containedGreen {
    color: white;
  }
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
`;
