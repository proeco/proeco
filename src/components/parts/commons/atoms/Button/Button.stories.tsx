import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Box } from '@mui/system';
import { Button } from './Button';

export default {
  title: 'parts/commons/atoms/Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => (
  <Box>
    <Box display="flex" alignItems="start" gap="20px" mb="30px">
      <Box>
        <Button size="small" color="primary" {...args}>
          Button!
        </Button>
      </Box>
      <Box>
        <Button size="medium" color="primary" {...args}>
          Button!
        </Button>
      </Box>
      <Box>
        <Button size="large" color="primary" {...args}>
          Button!
        </Button>
      </Box>
    </Box>
    <Box display="flex" alignItems="start" gap="20px">
      <Box>
        <Button size="small" color="secondary" {...args}>
          Button!
        </Button>
      </Box>
      <Box>
        <Button size="medium" color="secondary" {...args}>
          Button!
        </Button>
      </Box>
      <Box>
        <Button size="large" color="secondary" {...args}>
          Button!
        </Button>
      </Box>
    </Box>
  </Box>
);

export const Contained = Template.bind({});
Contained.args = {
  variant: 'contained',
  bold: true,
};

export const Outlined = Template.bind({});
Outlined.args = {
  variant: 'outlined',
  bold: true,
};

export const Text = Template.bind({});
Text.args = {
  variant: 'text',
  bold: true,
};
