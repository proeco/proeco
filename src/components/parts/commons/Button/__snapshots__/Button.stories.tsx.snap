// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots parts/commons/Button Contained 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root css-mkotpv-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1brzon9-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-1exi219-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root css-ze7b1j-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1fyly1l-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-1gxgd1x-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedGreen MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root css-15mz20-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedGreen MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-m382e3-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedGreen MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-jsp0u2-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-q0kkji"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root css-1nmush4-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1h25v82-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-14lyiag-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots parts/commons/Button Outlined 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-ejj0pz-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-1enn4hz-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root css-d0pwjv-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-oa183p-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-16fgh9r-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root css-16bmqb0-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedGreen MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-17mqm23-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedGreen MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-1j0kw0x-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedGreen MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root css-3zggkf-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-q0kkji"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root css-1545cp5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-1751hni-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root css-11308dp-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots parts/commons/Button Text 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-8t3ak0-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-rf4qge-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root css-3dm46d-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-1m28ka8-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-p2i07q-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root css-14ihq79-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-1h74uyw"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textGreen MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-mn2eqa-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textGreen MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1og53vp-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textGreen MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root css-j389y9-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root css-q0kkji"
  >
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-32yj5h-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-606krt-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <button
        bold={1}
        className="MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root css-1vxbond-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Button!
      </button>
    </div>
  </div>
</div>
`;
